c7n/tags.py

```diff
@@ -1032,15 +1054,26 @@ def validate(self):
 
     def process(self, resources):
         related_resources = []
-        for rrid, r in zip(jmespath.search('[].[%s]' % self.data['key'], resources),
+        search_path = "[].[%s]" % self.data['key']
```

Comment:
nit: looks like this could go in an else below, since it'll be redefined for the `tag:` case